### PR TITLE
feat: support crashpad on linux

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This enables crashpad support on Linux via `--enable-crashpad`. ~~Currently this
must be supplied on the command line and cannot be enabled through
`app.commandLine.appendSwitch`, because the zygote process on Linux is launched
prior to the first JS execution in the main process.~~ enabling it from JS seems to work now?

The plan is to ultimately remove this switch and only support crashpad.
Upstream, Chromium has already enabled crashpad by default on Linux:
https://crrev.com/83bf8a5bbe7f9af4f4531b65662cfeb0c232b583

Landing behind a flag first in order to match upstream, and to make it easily
revertable once we enable it by default.

No release notes as this does not change behavior by default, and I don't think
the `--enable-crashpad` flag is something we especially want to advertise.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added

#### Release Notes

Notes: none

### CODE REVIEW

File: shell/browser/electron_browser_client.cc [hunk 1]
```diff
@@ -297,6 +297,12 @@ breakpad::CrashHandlerHostLinux* CreateCrashHandlerHost(
 }
 
 int GetCrashSignalFD(const base::CommandLine& command_line) {
+  if (crash_reporter::IsCrashpadEnabled()) {
+    int fd;
+    pid_t pid;
+    return crash_reporter::GetHandlerSocket(&fd, &pid) ? fd : -1;
```
Comment (ckerr) [inline shell/browser/electron_browser_client.cc:303]: _micronit_ we can pass a nullptr pid here

**Comment by nornagon** [comment]
Linux/ia32 tests blocked on https://github.com/electron/build-images/pull/10

File: shell/browser/electron_browser_client.cc (hunk 1 above)
Comment (nornagon) [reply shell/browser/electron_browser_client.cc:304]: True, though this is copy/pasted from Chrome: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_content_browser_client.cc;l=968;drc=161239673fd3fac73ecbf3d4eedb770f3e0d7fa2

File: shell/browser/api/electron_api_crash_reporter.cc [hunk 2]
```diff
@@ -203,7 +217,20 @@ scoped_refptr<UploadList> CreateCrashUploadList() {
...
       crash_dir_path.AppendASCII(CrashUploadList::kReporterLogFilename);
-  return base::MakeRefCounted<TextLogUploadList>(upload_log_path);
+  scoped_refptr<UploadList> result =
+      base::MakeRefCounted<TextLogUploadList>(upload_log_path);
+  if (crash_reporter::IsCrashpadEnabled()) {
+    // Crashpad keeps the records of C++ crashes (segfaults, etc) in its
+    // internal database. The JavaScript error reporter writes JS error upload
+    // records to the older text format. Combine the two to present a complete
+    // list to the user.
+    // TODO(nornagon): what is "The JavaScript error reporter", and do we care
```
Comment (deepak1556) [inline shell/browser/api/electron_api_crash_reporter.cc:227]: https://bugs.chromium.org/p/chromium/issues/detail?id=1121816 , looks like it is used to capture JS errors from webui code. Implementation looks quite simple, they hook onto the console message observer to collect data https://source.chromium.org/chromium/chromium/src/+/main:content/browser/webui/web_ui_main_frame_observer.cc;l=71

**Review by deepak1556 — APPROVED** [review]
LGTM

**Review by ckerr — APPROVED** [review]
Looks good. Thanks for doing this!

**Review by jkleinsc — CHANGES_REQUESTED** [review]
Looks like the gn check is failing on MAS: 
https://app.circleci.com/pipelines/github/electron/electron/42346/workflows/61f131e3-144e-4f9b-b6dd-7c15f109e06d/jobs/945470

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**